client/src/about_page_components/AboutNavigation.js

```diff
@@ -0,0 +1,146 @@
+import React, { useState, useEffect, useRef } from "react";
+import { Slider } from "@mui/material";
+import {
+  IoPlaySkipBack,
+  IoPlaySkipForward,
+  IoPlay,
+  IoPause,
+} from "react-icons/io5";
+
+function generate_markers(val, scale) {
+  let res = [];
+  for (let i = 0; i < val; i++) {
+    res.push({ value: i * scale });
+  }
+  return res;
+}
+
+function useInterval(callback, delay) {
+  const savedCallback = useRef();
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  useEffect(() => {
+    function tick() {
+      savedCallback.current();
+    }
+
+    if (delay !== null) {
+      let id = setInterval(tick, delay);
+      return () => clearInterval(id);
+    }
+  }, [delay]);
+}
+
+const AboutNavigation = () => {
+  const [mark, setMark] = useState([]);
+  const [run, setRun] = useState(false);
+  const [sliderVal, setSliderVal] = useState(0);
+  const [counter, setCounter] = useState(0);
+
+  let scale = 100;
+
+  useInterval(
+    () => {
+      if (sliderVal > 10 * scale) {
+        setRun(false);
+      }
+
+      setSliderVal(sliderVal + 1);
+      setCounter(Math.floor(sliderVal / scale));
+    },
+    run ? 50 : null
+  );
+
+  useEffect(() => {
+    setMark(generate_markers(10, scale));
+  }, [10]);
+
+  useEffect(() => {
+    setSliderVal(counter * scale);
+  }, [counter]);
+
+  const onNext = () => {
+    if (counter === 10) {
+      return;
+    }
+
+    setCounter(counter + 1);
+  };
+
+  const onPrev = () => {
+    if (counter === 0) {
+      return;
+    }
+
+    setCounter(counter - 1);
+  };
+
+  return (
```

Comment:
(opt) I was wondering, would it be useful to have a small graph above the slider and the navigation buttons just to see how the graph changes? And maybe a 'number of characters' slider so people can see what the pruning looks like? Feel free to ignore

Reply:
Yeah, I did think of that but I would be basically rewriting the front end for a small section of a page so I opted for just the components. I will add the pruning slider once James has done his edits.